mx_fetchjdk.py

```diff
@@ -0,0 +1,204 @@
+#
+# ----------------------------------------------------------------------------------------------------
+#
+# Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+# ----------------------------------------------------------------------------------------------------
+#
+from __future__ import print_function
+import os, shutil, json
+from os.path import join, exists, abspath, isdir, islink
+from shutil import copytree
+from argparse import ArgumentParser, REMAINDER
+
+from mx import optional_suite_context, _mx_home, command
+from jdk_distribution_parser import JdkDistribution
+import mx, mx_urlrewrites
+
+@command('mx', 'fetch-jdk', '[options]')
+@optional_suite_context
+def fetch_jdk(args):
+    """fetches required JDK version
+
+    If called without --quiet flag, menu will be printed for available JDK selection.
+    """
+    args = _parse_fetchsettings(args)
+
+    distribution = args["java-distribution"]
+    base_path = args["base-path"]
+    artifact = distribution.get_folder_name()
+    final_path = distribution.get_final_path(base_path)
+    url = mx_urlrewrites.rewriteurl(distribution.get_url())
+    sha_url = url + ".sha1"
+    archive_name = distribution.get_archive_name()
+
+    if not args["quiet"] and not mx.ask_yes_no("Install {} to {}".format(artifact, final_path), default='y'):
+        mx.abort("JDK installation canceled")
+
+    if not exists(final_path):
+        if not args["quiet"]:
+            print("Fetching {} archive from {}...".format(artifact, url))
+
+        archive_location = join(base_path, archive_name)
+        mx._opts.no_download_progress = args["quiet"]
+        sha1_hash = mx._hashFromUrl(sha_url).decode('utf-8')
+
+        if not exists(archive_location) or not args["keep-archive"]:
```

Comment:
So `--keep` silently does nothing when the JDK is already installed? It should print a warning:
```
The --keep option is ignored when the JDK is already installed.
```
If `--keep` is specified and the archive is available, its location should be printed.